src/profiler/Elastic.Apm.Profiler.Managed.Loader/Logger.cs

```diff
@@ -5,134 +5,92 @@
-				return;
+	public static void Debug(string message, params object[] args) => Log(LogLevel.Debug, message, args);
 
-			Log(level, $"{message}{Environment.NewLine}{exception}", args);
-		}
+	public static void Error(Exception exception, string message, params object[] args) => Log(LogLevel.Error, exception, message, args);
 
-		public static void Log(LogLevel level, string message, params object[] args)
-		{
-			if (Level > level)
-				return;
+	public static void Error(string message, params object[] args) => Log(LogLevel.Error, message, args);
 
-			try
-			{
-				if (LogFile != null)
-				{
-					try
-					{
-						using (var stream = File.Open(LogFile, FileMode.Append, FileAccess.Write, FileShare.Read))
-						using (var writer = new StreamWriter(stream, new UTF8Encoding(false)))
-						{
-							writer.Write($"[{DateTimeOffset.Now:O}] [{Levels[level]}] ");
-							writer.WriteLine(message, args);
-							writer.Flush();
-							stream.Flush(true);
-						}
-
-						return;
-					}
-					catch
-					{
-						// ignore
-					}
-				}
-
-				Console.Error.WriteLine($"[{DateTimeOffset.Now:O}] [{Levels[level]}] {message}", args);
-			}
-			catch
-			{
-				// ignore
-			}
-		}
-
-		private static string GetLogFile(string logDirectory)
-		{
-			if (logDirectory is null)
-				return null;
+	public static void Log(LogLevel level, Exception exception, string message, params object[] args)
+	{
+		if (!IsActive || Level > level)
+			return;
 
-			var process = Process.GetCurrentProcess();
-			return Path.Combine(logDirectory, $"Elastic.Apm.Profiler.Managed.Loader_{process.ProcessName}_{process.Id}.log");
-		}
+		Log(level, $"{message}{Environment.NewLine}{exception}", args);
+	}
 
-		private static LogLevel GetLogLevel(LogLevel defaultLevel)
-		{
-			var level = Environment.GetEnvironmentVariable("ELASTIC_APM_PROFILER_LOG");
-			if (string.IsNullOrEmpty(level))
-				return defaultLevel;
+	public static void Log(LogLevel level, string message, params object[] args)
+	{
+		if (!IsActive || Level > level)
+			return;
 
-			return Enum.TryParse<LogLevel>(level, true, out var parsedLevel)
-				? parsedLevel
-				: defaultLevel;
-		}
+		if (string.IsNullOrWhiteSpace(LogFile))
+			return;
 
-		private static string GetLogDirectory()
+		try
 		{
 			try
 			{
-				var logDirectory = Environment.GetEnvironmentVariable("ELASTIC_APM_PROFILER_LOG_DIR");
-				if (string.IsNullOrEmpty(logDirectory))
-				{
-					if (Environment.OSVersion.Platform == PlatformID.Win32NT)
-					{
-						var programData = Environment.GetEnvironmentVariable("PROGRAMDATA");
-						logDirectory = !string.IsNullOrEmpty(programData)
-							? Path.Combine(programData, "elastic", "apm-agent-dotnet", "logs")
-							: ".";
-					}
-					else
-						logDirectory = "/var/log/elastic/apm-agent-dotnet";
-				}
-
-				Directory.CreateDirectory(logDirectory);
-				return logDirectory;
+				using var stream = File.Open(LogFile, FileMode.Append, FileAccess.Write, FileShare.Read);
```

Comment:
I'm not sure how opening the file for each log entry performs compared to holding the file open while the process is running. However, as this already exists, I guess we can keep it for now as we've not had any specific reports of issues.

Reply:
Aye not ideal however the loader is only short lived.